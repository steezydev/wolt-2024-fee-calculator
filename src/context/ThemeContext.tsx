import { Theme, ThemeContextType } from '@/types/shared/Theme';
import React, {
  ReactNode,
  createContext,
  useContext,
  useEffect,
  useState,
} from 'react';

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

export const useTheme = (): ThemeContextType => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};

export const ThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const [theme, setTheme] = useState<Theme>(Theme.LIGHT);

  useEffect(() => {
    const savedTheme = localStorage.getItem('theme') as Theme | null;
    const prefersDark = window.matchMedia(
      '(prefers-color-scheme: dark)'
    ).matches;
    const initialTheme = savedTheme || (prefersDark ? Theme.DARK : Theme.LIGHT);
    setTheme(initialTheme);
    document.documentElement.classList.toggle(
      'dark',
      initialTheme === Theme.DARK
    );
  }, []);

  const toggleTheme = (theme: Theme) => {
    setTheme(theme);
    localStorage.setItem('theme', theme);
    document.documentElement.classList.toggle('dark', theme === Theme.DARK);
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};
