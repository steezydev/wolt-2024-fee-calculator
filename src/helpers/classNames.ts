export const classNames = (...classes: (string | undefined | false)[]) => {
    return classes.filter(Boolean).join(' ');
};
