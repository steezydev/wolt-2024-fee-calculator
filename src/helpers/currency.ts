import { Currency } from '@/types/shared/Currency';

export const formatCurrency = (value: number, currency: Currency) => {
    return new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: currency,
        minimumFractionDigits: 2,
    }).format(value);
};
