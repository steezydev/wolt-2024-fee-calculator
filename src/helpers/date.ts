import { parse } from 'date-fns';

export const isValidDate = (dateStr: string): boolean => {
    const regex = /^(\d{2})\.(\d{2})\.(\d{4}) (\d{2}):(\d{2})$/;
    if (!regex.test(dateStr)) return false;

    // Get date components from string
    const [_, day, month, year, hour, minute] = regex
        .exec(dateStr)!
        .map(Number);
    const date = new Date(year, month - 1, day, hour, minute);

    return (
        date.getFullYear() === year &&
        date.getMonth() === month - 1 &&
        date.getDate() === day &&
        date.getHours() === hour &&
        date.getMinutes() === minute
    );
};

export const autoCompleteDate = (dateStr: string): string => {
    // Regular expression to match partial date-time inputs
    const regex = /^(\d{0,2})\.?(\d{0,2})\.?(\d{0,4}) ?(\d{0,2})?:?(\d{0,2})?$/;
    const matches = dateStr.match(regex);
    if (!matches) return dateStr;

    // Get date components from string
    let [_, day, month, year, hour, minute] = matches;

    // Current date as fallback
    const currentDate = new Date();
    const currentYear = currentDate.getFullYear().toString();
    const currentMonth = (currentDate.getMonth() + 1)
        .toString()
        .padStart(2, '0');
    const currentDay = currentDate.getDate().toString().padStart(2, '0');

    // Auto-complete year, month, and day
    year = year.padStart(
        4,
        year.length === 2 ? '20' : currentYear.slice(0, 4 - year.length)
    );
    month = month.padStart(2, currentMonth.slice(0, 2 - month.length));
    day = day.padStart(2, currentDay.slice(0, 2 - day.length));

    // Auto-complete hour and minute
    if (hour && hour.length === 1) {
        hour += '0'; // Add trailing zero if only one digit is entered
    } else if (hour === undefined || hour.length === 0) {
        hour = '00'; // Default to 00 if hour is empty
    }

    if (minute && minute.length === 1) {
        minute += '0'; // Add trailing zero if only one digit is entered
    } else if (minute === undefined || minute.length === 0) {
        minute = '00'; // Default to 00 if minute is empt
    }

    return `${day}.${month}.${year} ${hour}:${minute}`;
};

export const dateTimeStringToDate = (dateTimeString: string): Date => {
    return parse(dateTimeString, 'dd.MM.yyyy HH:mm', new Date());
};

export const setDateDate = (date: Date, newDate: Date): Date => {
    const newDateDate = new Date(date);
    newDateDate.setFullYear(newDate.getFullYear());
    newDateDate.setMonth(newDate.getMonth());
    newDateDate.setDate(newDate.getDate());

    return newDateDate;
};
