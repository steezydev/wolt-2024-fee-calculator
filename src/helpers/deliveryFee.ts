const baseFee = 2;
const extraItemSurcharge = 0.5;
const bulkSurcharge = 1.2;
const rushHourMultiplier = 1.2;
const freeDeliveryThreshold = 200;
const maxDeliveryFee = 15;

export function isFridayRushHour(date: Date): boolean {
    const day = date.getDay();
    const hour = date.getHours();

    return day === 5 && hour >= 15 && hour < 19;
}

export function calculateSmallOrderSurcharge(cartValue: number): number {
    // Convert to cents to avoid floating point errors
    const valueInCents = cartValue * 100;
    const surcharge = valueInCents < 1000 ? 1000 - valueInCents : 0;
    return surcharge / 100;
}

export function calculateDistanceFee(distance: number): number {
    if (distance <= 1000) return baseFee;
    return baseFee + Math.ceil((distance - 1000) / 500);
}

export function calculateItemSurcharge(itemCount: number): number {
    if (itemCount < 5) return 0;

    const surcharge = (itemCount - 4) * extraItemSurcharge;

    if (itemCount <= 12) return surcharge;

    return surcharge + bulkSurcharge;
}

export function calculateDeliveryFee(
    cartValue: number,
    distance: number,
    itemCount: number,
    orderTime: Date
): number {
    if (cartValue >= freeDeliveryThreshold) return 0; // Free delivery for cart value >= 200â‚¬

    let fee =
        calculateSmallOrderSurcharge(cartValue) +
        calculateDistanceFee(distance) +
        calculateItemSurcharge(itemCount);

    // Apply Friday rush hour multiplier
    if (isFridayRushHour(orderTime)) {
        fee *= rushHourMultiplier;
    }

    // Ensure the fee does not exceed the maximum limit
    return Math.min(fee, maxDeliveryFee);
}
