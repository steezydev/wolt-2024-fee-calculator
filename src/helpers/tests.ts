const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
