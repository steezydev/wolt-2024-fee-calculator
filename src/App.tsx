import ButtonPrimary from '@/components/button/ButtonPrimary';
import BadgeInfoIcon from '@/components/icons/BadgeInfoIcon';
import InputRegular from '@/components/input/InputRegular';
import InputLabel from '@/components/inputlabel/InputLabel';
import B2 from '@/components/typography/B2';
import H1 from '@/components/typography/H1';
import { wihtInputIcon } from '@/hoc/InputHoc';
import { format } from 'date-fns';
import { useState } from 'react';

import CalculationResult from './components/calculationresult/CalculationResult';
import CalculationResultItem from './components/calculationresult/CalculationResultItem';
import DateTimePicker from './components/datetimepicker/DateTimePicker';
import CalendarClockIcon from './components/icons/CalendarClockIcon';
import EuroIcon from './components/icons/EuroIcon';
import MapPinIcon from './components/icons/MapPinIcon';
import ThemeSwitch from './components/themeswitch/ThemeSwitch';
import { calculateDeliveryFee } from './helpers/deliveryFee';
import { withThemeSelectorHoc } from './hoc/ThemeHoc';
import useInput from './hooks/useInput';

//TODO: Write comments for components (everything)

const CartValueInput = wihtInputIcon(InputRegular, EuroIcon);
const DeliveryDistanceInput = wihtInputIcon(InputRegular, MapPinIcon);
const ItemsAmountInput = InputRegular;
const OrderTimeInput = wihtInputIcon(InputRegular, CalendarClockIcon);
const ThemeSwitchSelector = withThemeSelectorHoc(ThemeSwitch);

const App = () => {
  const cartValueInputData = useInput('', {
    required: true,
    type: 'numeric',
  });

  const deliveryDistanceInputData = useInput('', {
    required: true,
    type: 'integer',
  });

  const itemsAmountInputData = useInput('', {
    required: true,
    type: 'integer',
  });

  const [orderTime, setOrderTime] = useState(new Date());
  const [deliveryPrice, setDeliveryPrice] = useState<number | null>(null);

  const isButtonActive =
    cartValueInputData.isValid &&
    cartValueInputData.isFilled &&
    deliveryDistanceInputData.isValid &&
    deliveryDistanceInputData.isFilled &&
    itemsAmountInputData.isValid &&
    itemsAmountInputData.isFilled &&
    orderTime;

  const handleFormSubmit = () => {
    const cartValue = parseFloat(cartValueInputData.value);
    const deliveryDistance = parseInt(deliveryDistanceInputData.value);
    const itemsAmount = parseInt(itemsAmountInputData.value);
    const orderDatetime = new Date(orderTime);

    console.log(cartValue, deliveryDistance, itemsAmount, orderDatetime);
    const fee = calculateDeliveryFee(
      cartValue,
      deliveryDistance,
      itemsAmount,
      orderDatetime
    );
    setDeliveryPrice(fee);
  };

  return (
    <>
      <header className='py-8 sm:py-12 flex flex-row items-start justify-between gap-5'>
        <H1 className='text-black dark:text-white'>Delivery fee calculator</H1>
        <ThemeSwitchSelector />
      </header>
      <main className='flex flex-col gap-6'>
        <div className='flex flex-row gap-3 items-center'>
          <BadgeInfoIcon className='text-primary-300' />
          <B2 className='text-black dark:text-white'>
            Fill the fields to calculate the delivery fee
          </B2>
        </div>
        <form className='flex flex-col gap-4 sm:gap-6'>
          <InputLabel
            label='Cart value, €'
            id='cartValueInput'
            errorMessage={cartValueInputData.error}
          >
            <CartValueInput
              required
              className='w-full'
              id='cartValueInput'
              name='cartValueInput'
              type='text'
              ariaLabel='Cart value in euros'
              placeholder='20'
              min={1}
              onBlur={cartValueInputData.handleTouch}
              isInvalid={!cartValueInputData.isValid}
              onChange={cartValueInputData.handleChange}
              value={cartValueInputData.value}
            />
          </InputLabel>
          <InputLabel
            label='Delivery distance, m'
            id='deliveryDistanceInput'
            errorMessage={deliveryDistanceInputData.error}
          >
            <DeliveryDistanceInput
              required
              className='w-full'
              id='deliveryDistanceInput'
              name='deliveryDistanceInput'
              type='text'
              ariaLabel='Delivery distance in meters'
              placeholder='900'
              min={0}
              onBlur={deliveryDistanceInputData.handleTouch}
              isInvalid={!deliveryDistanceInputData.isValid}
              onChange={deliveryDistanceInputData.handleChange}
              value={deliveryDistanceInputData.value}
            />
          </InputLabel>
          <InputLabel
            label='Amount of items'
            id='deliveryItemsAmountInput'
            errorMessage={itemsAmountInputData.error}
          >
            <ItemsAmountInput
              required
              className='w-full'
              id='deliveryItemsAmountInput'
              name='deliveryItemsAmountInput'
              type='text'
              ariaLabel='Amount of items in cart'
              placeholder='1'
              min={1}
              onBlur={itemsAmountInputData.handleTouch}
              isInvalid={!itemsAmountInputData.isValid}
              onChange={itemsAmountInputData.handleChange}
              value={itemsAmountInputData.value}
            />
          </InputLabel>
          <InputLabel
            label='Order time'
            id='deliveryOrderDateTimeInput'
            errorMessage='Please enter a valid date'
          >
            <DateTimePicker
              Item={OrderTimeInput}
              required
              className='w-full'
              id='deliveryOrderDateTimeInput'
              name='deliveryOrderDateTimeInput'
              ariaLabel='Order time'
              placeholder='hh.mm.yyyy HH:MM'
              onChange={setOrderTime}
              value={orderTime}
            />
          </InputLabel>
          <ButtonPrimary
            id='submitCalculateDeliveryPrice'
            ariaLabel='Calculate delivery price'
            disabled={!isButtonActive}
            className='mt-4 w-full'
            onClick={handleFormSubmit}
          >
            Calculate delivery price
          </ButtonPrimary>
        </form>
        {deliveryPrice !== null && (
          <CalculationResult result={deliveryPrice}>
            <CalculationResultItem
              label='Cart value'
              value={`${cartValueInputData.value} €`}
            />
            <CalculationResultItem
              label='Delivery distance'
              value={`${deliveryDistanceInputData.value} m`}
            />
            <CalculationResultItem
              label='Amount of items'
              value={itemsAmountInputData.value}
            />
            <CalculationResultItem
              label='Order time'
              value={format(orderTime, 'dd.MM.yyyy HH:mm')}
            />
          </CalculationResult>
        )}
      </main>
    </>
  );
};

export default App;
