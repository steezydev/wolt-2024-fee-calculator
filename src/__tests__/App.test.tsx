import App from '@/App';
import { ThemeProvider } from '@/context/ThemeContext';
import { calculateDeliveryFee } from '@/helpers/deliveryFee';
import { mockContext, setupFakeDate } from '@/helpers/tests';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';

jest.mock('@/helpers/deliveryFee');

mockContext();
setupFakeDate();

describe('App', () => {
  it('renders', () => {
    const component = TestRenderer.create(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );
    expect(screen.getByText('Delivery fee calculator')).toBeInTheDocument();
    expect(screen.getByLabelText('Main content')).toBeInTheDocument();
    expect(screen.getByText('Order fee calculation form')).toBeInTheDocument();
  });

  it('handles form submission', async () => {
    const mockFee = 10;
    (calculateDeliveryFee as jest.Mock).mockReturnValue(mockFee);

    render(
      <ThemeProvider>
        <App />
      </ThemeProvider>
    );

    fireEvent.change(screen.getByLabelText('Cart value in euros'), {
      target: { value: '20' },
    });
    fireEvent.change(screen.getByLabelText('Delivery distance in meters'), {
      target: { value: '5' },
    });
    fireEvent.change(screen.getByLabelText('Amount of items in cart'), {
      target: { value: '3' },
    });
    fireEvent.change(screen.getByLabelText('Order time'), {
      target: { value: '01.01.2024 00:00' },
    });

    fireEvent.click(screen.getByText('Calculate delivery price'));

    expect(calculateDeliveryFee).toHaveBeenCalledWith(
      20,
      5,
      3,
      new Date('2024-01-01')
    );

    expect(screen.getByText(`â‚¬${mockFee}.00`)).toBeInTheDocument();
  });
});
