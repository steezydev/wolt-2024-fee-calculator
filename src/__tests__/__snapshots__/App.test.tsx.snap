// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
[
  <header
    aria-labelledby="header"
    className="py-8 sm:py-12 flex flex-row items-start justify-between gap-5"
  >
    <h1
      className="font-display text-4xl sm:text-5xl font-bold subpixel-antialiased text-black dark:text-white"
      id="header"
    >
      Delivery fee calculator
    </h1>
    <div
      className="flex gap-3"
    >
      <button
        aria-label="Set light theme"
        className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none bg-primary-300 text-white"
        data-test-id="light-theme-button"
        id="light-theme-button"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="4"
          />
          <path
            d="M12 2v2"
          />
          <path
            d="M12 20v2"
          />
          <path
            d="m4.93 4.93 1.41 1.41"
          />
          <path
            d="m17.66 17.66 1.41 1.41"
          />
          <path
            d="M2 12h2"
          />
          <path
            d="M20 12h2"
          />
          <path
            d="m6.34 17.66-1.41 1.41"
          />
          <path
            d="m19.07 4.93-1.41 1.41"
          />
        </svg>
      </button>
      <button
        aria-label="Set dark theme"
        className="rounded-full p-2 text-black dark:text-white hover:bg-primary-300/90 hover:text-white active:bg-primary-300/80 disabled:bg-primary-300/60 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none"
        data-test-id="dark-theme-button"
        id="dark-theme-button"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
          />
        </svg>
      </button>
    </div>
  </header>,
  <main
    aria-label="Main content"
    className="flex flex-col gap-6"
  >
    <div
      className="flex flex-row gap-3 items-center"
    >
      <svg
        aria-hidden="true"
        className="text-primary-300"
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.85 8.62a4 4 0 0 1 4.78-4.77 4 4 0 0 1 6.74 0 4 4 0 0 1 4.78 4.78 4 4 0 0 1 0 6.74 4 4 0 0 1-4.77 4.78 4 4 0 0 1-6.75 0 4 4 0 0 1-4.78-4.77 4 4 0 0 1 0-6.76Z"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
      <p
        className="text-lg text-black dark:text-white"
        id="orderFeeFormDescription"
      >
        Order fee calculation form
      </p>
    </div>
    <form
      aria-labelledby="orderFeeFormDescription"
      className="flex flex-col gap-4 sm:gap-6"
      onSubmit={[Function]}
    >
      <div
        className="group flex flex-col gap-1"
      >
        <div
          className="relative items-center inline-block peer order-2 w-full"
          data-focused={false}
          data-invalid={false}
        >
          <input
            aria-disabled={false}
            aria-errormessage="cartValueInputErrorMessage"
            aria-invalid={false}
            aria-label="Cart value in euros"
            aria-required={true}
            autoFocus={false}
            className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:placeholder:text-error data-[invalid=true]:dark:placeholder:text-error data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300 pl-10 w-full peer"
            data-invalid={false}
            data-test-id="cartValue"
            disabled={false}
            id="cartValue"
            min={1}
            name="cartValueInput"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Cart value in euros..."
            readOnly={false}
            required={true}
            type="text"
            value=""
          />
          <div
            className="absolute left-2 top-0 h-full flex items-center text-neutral-light-300 dark:text-neutral-dark-300 peer-data-[invalid=true]:text-error peer-focus:text-primary-300"
          >
            <svg
              aria-hidden="true"
              className="text-inherit"
              fill="none"
              height="24"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 10h12"
              />
              <path
                d="M4 14h9"
              />
              <path
                d="M19 6a7.7 7.7 0 0 0-5.2-2A7.9 7.9 0 0 0 6 12c0 4.4 3.5 8 7.8 8 2 0 3.8-.8 5.2-2"
              />
            </svg>
          </div>
        </div>
        <label
          className="text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1 peer-focus:text-primary-300 peer-data-[invalid=true]:text-error peer-data-[focused=true]:peer-data-[invalid=false]:text-primary-300"
          htmlFor="cartValue"
        >
          Cart value, â‚¬
        </label>
      </div>
      <div
        className="group flex flex-col gap-1"
      >
        <div
          className="relative items-center inline-block peer order-2 w-full"
          data-focused={false}
          data-invalid={false}
        >
          <input
            aria-disabled={false}
            aria-errormessage="deliveryDistanceInputErrorMessage"
            aria-invalid={false}
            aria-label="Delivery distance in meters"
            aria-required={true}
            autoFocus={false}
            className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:placeholder:text-error data-[invalid=true]:dark:placeholder:text-error data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300 pl-10 w-full peer"
            data-invalid={false}
            data-test-id="deliveryDistance"
            disabled={false}
            id="deliveryDistance"
            min={0}
            name="deliveryDistanceInput"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Delivery distance in meters..."
            readOnly={false}
            required={true}
            type="text"
            value=""
          />
          <div
            className="absolute left-2 top-0 h-full flex items-center text-neutral-light-300 dark:text-neutral-dark-300 peer-data-[invalid=true]:text-error peer-focus:text-primary-300"
          >
            <svg
              aria-hidden="true"
              className="text-inherit"
              fill="none"
              height="24"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 10c0 6-8 12-8 12s-8-6-8-12a8 8 0 0 1 16 0Z"
              />
              <circle
                cx="12"
                cy="10"
                r="3"
              />
            </svg>
          </div>
        </div>
        <label
          className="text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1 peer-focus:text-primary-300 peer-data-[invalid=true]:text-error peer-data-[focused=true]:peer-data-[invalid=false]:text-primary-300"
          htmlFor="deliveryDistance"
        >
          Delivery distance, m
        </label>
      </div>
      <div
        className="group flex flex-col gap-1"
      >
        <input
          aria-disabled={false}
          aria-errormessage="deliveryItemsAmountInputErrorMessage"
          aria-invalid={false}
          aria-label="Amount of items in cart"
          aria-required={true}
          autoFocus={false}
          className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:placeholder:text-error data-[invalid=true]:dark:placeholder:text-error data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300 peer order-2 w-full"
          data-invalid={false}
          data-test-id="numberOfItems"
          disabled={false}
          id="numberOfItems"
          min={1}
          name="deliveryItemsAmountInput"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Number of items..."
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <label
          className="text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1 peer-focus:text-primary-300 peer-data-[invalid=true]:text-error peer-data-[focused=true]:peer-data-[invalid=false]:text-primary-300"
          htmlFor="numberOfItems"
        >
          Amount of items
        </label>
      </div>
      <div
        className="group flex flex-col gap-1"
      >
        <div
          className="w-full relative peer order-2"
          data-focused={false}
          data-invalid={false}
        >
          <div
            className="flex items-stretch gap-2"
          >
            <div
              className="relative items-center inline-block w-full"
              data-focused={false}
            >
              <input
                aria-disabled={false}
                aria-errormessage="deliveryOrderDateTimeInputErrorMessage"
                aria-invalid={false}
                aria-label="Order time"
                aria-required={true}
                autoFocus={false}
                className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none rounded-xl border-2 bg-transparent border-neutral-light-100 placeholder:text-neutral-light-300 px-3 py-2.5 text-xl text-black leading-none  dark:border-neutral-dark-100 dark:placeholder:text-neutral-dark-300 dark:text-white group-hover:data-[invalid=false]:border-primary-300 data-[invalid=true]:placeholder:text-error data-[invalid=true]:dark:placeholder:text-error data-[invalid=true]:border-error data-[invalid=true]:dark:border-error focus:data-[invalid=true]:border-primary-300 focus:data-[invalid=true]:dark:border-primary-300 focus:outline-none focus:ring-1 focus:ring-primary-300 focus:border-primary-300 focus:dark:border-primary-300 pl-10 w-full peer"
                data-invalid={false}
                data-test-id="orderTime"
                disabled={false}
                id="orderTime"
                name="deliveryOrderDateTimeInput"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Date and time... (hh.mm.yyyy HH:MM)"
                readOnly={false}
                required={true}
                type="text"
                value="01.01.2024 03:00"
              />
              <div
                className="absolute left-2 top-0 h-full flex items-center text-neutral-light-300 dark:text-neutral-dark-300 peer-data-[invalid=true]:text-error peer-focus:text-primary-300"
              >
                <svg
                  aria-hidden="true"
                  className="text-inherit"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 7.5V6a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h3.5"
                  />
                  <path
                    d="M16 2v4"
                  />
                  <path
                    d="M8 2v4"
                  />
                  <path
                    d="M3 10h5"
                  />
                  <path
                    d="M17.5 17.5 16 16.25V14"
                  />
                  <path
                    d="M22 16a6 6 0 1 1-12 0 6 6 0 0 1 12 0Z"
                  />
                </svg>
              </div>
            </div>
            <button
              aria-label="Open datepicker modal"
              className="bg-primary-100 text-lg font-bold rounded-xl hover:bg-primary-300/20 active:bg-primary-300/50 disabled:bg-neutral-light-100 disabled:dark:bg-neutral-dark-100 disabled:text-white/40 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none aspect-square flex justify-center items-center text-primary-300 px-3.5"
              data-test-id="openDatePickerButton"
              disabled={false}
              id="openDatePickerButton"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                className="w-6 h-6 text-inherit"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 2v4"
                />
                <path
                  d="M16 2v4"
                />
                <rect
                  height="18"
                  rx="2"
                  width="18"
                  x="3"
                  y="4"
                />
                <path
                  d="M3 10h18"
                />
                <path
                  d="M10 16h4"
                />
                <path
                  d="M12 14v4"
                />
              </svg>
            </button>
          </div>
        </div>
        <label
          className="text-neutral-light-300 dark:text-neutral-dark-300 text-sm font-semibold order-1 peer-focus:text-primary-300 peer-data-[invalid=true]:text-error peer-data-[focused=true]:peer-data-[invalid=false]:text-primary-300"
          htmlFor="orderTime"
        >
          Order time
        </label>
      </div>
      <button
        aria-label="Calculate delivery price"
        className="bg-primary-300 text-white text-lg font-bold py-3 rounded-xl hover:bg-primary-300/90 active:bg-primary-300/80 disabled:bg-neutral-400 disabled:dark:bg-neutral-700 disabled:text-white/60 disabled:cursor-not-allowed focus-visible:ring-4 focus-visible:ring-focus focus:outline-none focus-visible:outline-none mt-4 w-full"
        data-test-id="submitCalculateDeliveryPrice"
        disabled={true}
        id="submitCalculateDeliveryPrice"
        type="submit"
      >
        Calculate delivery price
      </button>
    </form>
  </main>,
]
`;
